referenceProvider.js

```diff
@@ -0,0 +1,128 @@
+        let moduleList = {};
+
+        if(range) {
+            const textAtCarrot = document.getText(range);
+            const fullTextWithoutComments = parser.removeComments(fullText);
+            const requireOrDefineStatement = parser.getRequireOrDefineStatement(fullTextWithoutComments);
+
+            if(requireOrDefineStatement) {
+                moduleList = parser.getModulesWithPathFromRequireOrDefine(requireOrDefineStatement);
+            }
+
+            const modulePath = textAtCarrot in moduleList ? moduleList[textAtCarrot] : null;
+
+            //We matched a module (word is a module)
+            if(modulePath) {
+                let searchFor = '';
+                let stopSearchingFurther;
+
+                if(childWord == "") {//Not a parent - search for the module name (word)
+                    searchFor = textAtCarrot;
+                    stopSearchingFurther = true;
+                } else { //It is a parent, search for the child which is a property of the module
+                    searchFor = childWord;
+                    stopSearchingFurther = false;
+                }
+                childWord = "";
+
+                return search.forModule(currentFilePath, modulePath, searchFor, stopSearchingFurther);
+            } else { //word is not a module
+                return new Promise(resolve => {
+                    let continueFrom;
+
+                    let dotPosition = document.offsetAt(new vscode.Position(range._start._line, range._start._character-1));
+
+                    //Do backwards search for a dot
+                    dotPosition = fullTextWithoutComments.lastIndexOf('.', dotPosition);
+                    const haveParent = dotPosition > 0;
+
+                    let tmpModuleName;
+                    if(!haveParent) {
+                        tmpModuleName = parser.extractString(document, range)
+                    }
+
+                    const constructors = parser.findConstructor(document, textAtCarrot, fullTextWithoutComments);
+                    //TODO: also consider window. defined globals
+                    //Dont have a parent and have a constructor, follow the constructor
+                    if(constructors.length && !haveParent) {
+                        //Break search in case the instance and the constructor have the same name
+                        if(document.getText(document.getWordRangeAtPosition(constructors[0].range._start)) == textAtCarrot) {
+                            resolve(undefined);
+                            return;
+                        } else {
+                            continueFrom = constructors[0].range._start;
+                        }
+                    } else if(haveParent) { //Have a parent - follow it
+                        const propertyParentPosition = document.positionAt(dotPosition);
+                        let bracketPosition = document.offsetAt(propertyParentPosition);
+                        //Do backwards search for a ")"
```

Comment:
The old method also returns false if there is a character other than a space or tab or new line. Are you sure this matches the same functionality? I doubt it.